velox/exec/tests/RowContainerTest.cpp

```diff
@@ -304,23 +346,147 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
 
     std::vector<std::optional<T>> expectedOrder = {
         std::nullopt, lowest, 0.0, min, max, inf, nan};
+    setupFloatTestExpectedOutput<std::vector<std::optional<T>>>(
+        expectedOrder, ascending, nullsFirst);
+    VectorPtr expected = makeNullableFlatVector<T>(expectedOrder);
+
+    // Verify compare method with two rows as input
+    {
+      std::sort(rows.begin(), rows.end(), [&](const char* l, const char* r) {
+        return rowContainer->compare(l, r, 0, {nullsFirst, ascending}) < 0;
+      });
+      VectorPtr result = BaseVector::create(type, numRows, pool_.get());
+      rowContainer->extractColumn(rows.data(), numRows, 0, result);
+      assertEqualVectors(expected, result);
+    }
+
+    // Verify compare method with row and decoded vector as input
+    {
+      std::sort(
```

Comment:
This usage of std::sort is very strange since comparison function usually compares two entries from the collector being sorted, but here we compare an element from a RowContainer with a value from a DecodedVector. This happens to work that Each row in a RowContainer has an equivalent row in a DecodedVector, but it is confusing to reader.

Let's find a different, less confusing, way to test comparison between RowContainter row and DecodedVector row.

Reply:
Right. The idea here was - with the assumption that decodedVector and the rows have the same order - to let the `sort` function do the comparison to test all value combinations.
I see how the initial author used this to write a short test that tests the specific compare API.
Regarding a replacement we could set up two input vectors (one for the row container, one for the decoded vector) and call the compare API. However, as expected output the result matrix would need to be generated based on the vector contents which includes implementing a function to do the value comparisons properly.